### ISSUE TITLE
[Bug]: Infinite synchronization loop for Strimzi CRDs in ArgoCD

### ISSUE THREAD

**Issue by jakubmalek** [strimzi/strimzi-kafka-operator#10541]
### Bug Description

There seem to be an issue with installation of Strimzi CRD manifests using ArgoCD. 
Several of the Strimzi CRD manifests declares an empty list of required properties in the JSON schema.

For example:
* https://github.com/strimzi/strimzi-kafka-operator/blob/0.43.0/helm-charts/helm3/strimzi-kafka-operator/crds/046-Crd-kafkabridge.yaml#L1101

```yaml
oneOf:
  - properties:
      secret: {}
      configMap: {}
      emptyDir: {}
      persistentVolumeClaim: {}
    required: []
```

The effective manifests in Kubernetes however ignores it resulting in

```yaml
oneOf:
  - properties:
      secret: {}
      configMap: {}
      emptyDir: {}
      persistentVolumeClaim: {}
```

without the `required: []` attribute.

This in consequence leads to the difference in the desired manifests in ArgoCD and infinite synchronization loop:
![image](https://github.com/user-attachments/assets/c50e30d5-bea3-487f-a7e5-c486d91d95f0)

Although technically, this is not a bug in Strimzi CRD manifests, the added attributes of empty list required properties doesn't add any value, and it makes the integration with ArgoCD more complicated.

Here is a list of the affected CRDs:
* `kafkabridges.kafka.strimzi.io`
* `kafkaconnects.kafka.strimzi.io`
* `kafkamirrormaker2s.kafka.strimzi.io`
* `kafkamirrormakers.kafka.strimzi.io`
* `kafkanodepools.kafka.strimzi.io`
* `kafkas.kafka.strimzi.io`

### Steps to reproduce

Install Strimzi Kafka Operator 0.43 with ArgoCD.

### Expected behavior

Installation is successful and all resources are in-sync.

### Strimzi version

0.43

### Kubernetes version

1.30

### Installation method

Helm chart, ArgoCD

### Infrastructure

EKS

**Comment by scholzj** [comment]
This was already addressed in
https://github.com/strimzi/strimzi-kafka-operator/pull/10492

**Comment by scholzj** [comment]
Discussed on community call on 5.9.2024: Fixed in #10492 (the PR also contains some possible workaround for ArgoCD).
